sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -972,11 +1024,46 @@ async def settle_messages_async(self, delivery_id: Union[int, Tuple[int, int]],
         except TypeError:
             first = delivery_id
             last = None
+
+        # Create a Message Delivery object for the Disposition
+        message_delivery = _MessageDelivery(
+            message,
+            MessageDeliveryState.WaitingToBeSent,
+            expire_time
+        )
+        on_disposition_received = partial(self._on_disposition_received_async, message_delivery)
+
         await self._link.send_disposition(
             first_delivery_id=first,
             last_delivery_id=last,
-            settled=True,
+            settled=False,
             delivery_state=state,
             batchable=batchable,
-            wait=True
+            wait=True,
+            message_delivery=message_delivery,
+            on_disposition = on_disposition_received,
         )
+
+        # Wait for the message to be settled
+        running = True
+        while running and message_delivery.state not in MESSAGE_DELIVERY_DONE_STATES:
+            # TODO: Confirm if this is the correct way to handle the do_work call
+            # calling do_work directly triggers Flow() calls which is not what we want
+            await self.do_work_async(flow=False)
```

Comment:
In case you need to break out using `running`. The loop could keep on running if the delivery state doesnt update but something else goes wrong. 

```suggestion
            running = await self.do_work_async(flow=False)
```